### DISCUSSION TITLE
第三方登录无法使用

### DISCUSSION THREAD

**Discussion by LxnChan** [walinejs/waline#966] [Q&A] [answered]
## 基本情况
 - 地址：https://lxnchan.cn
 - 部署方式：云函数CloudBase
 - 出问题的页面：任一文章页面下方评论
## 问题
登录时使用任意第三方登录（Github、Twitter、Facebook）都会返回404，可能是回调地址有问题。
## 另外的建议
建议自定义添加其他授权登录方式（微信、QQ等），同时也建议将token开放出来（让部署者自行申请对应授权平台的Key和Token）。

**Comment by lizheming** [comment, marked as answer]
同 https://github.com/walinejs/waline/issues/922 
CloudBase 子路径的方式 Waline 没办法识别到子路径，要么像 #922 一样定义根路径的，要不就是在服务端定义一个 `SERVER_URL` 变量也可以。

第三方登录可以自行部署 https://github.com/walinejs/auth 然后配置 `OAUTH_URL` 即可，参考文档 https://waline.js.org/reference/server.html#%E4%B8%BB%E8%A6%81%E9%85%8D%E7%BD%AE

**Reply by LxnChan** [reply]
谢谢。
另外请问自行部署的Auth只能部署在Vercel上吗？可以部署在腾讯云开发或本机部署吗？自行部署的Auth环境又是什么呢？

**Reply by lizheming** [reply]
目前只有 Vercel 的部署方式，如果需要部署腾讯云开发的话可以自行研究一下
